Resources/doc/first_steps.rst

```diff
@@ -0,0 +1,456 @@
+    {
+        public function createProduct(DocumentManager $dm)
+        {
+            $product = new Product();
+            $product->setName('A Foo Bar');
+            $product->setPrice('19.99');
+
+            $dm->persist($product);
+            $dm->flush();
+
+            return new Response('Created product id '.$product->getId());
+        }
+    }
+
+.. note::
+
+    In fact, since Doctrine is aware of all your managed objects, when you
+    call the ``flush()`` method, it calculates an overall changeset and executes
+    the most efficient operation possible.
+
+When creating or updating objects, the workflow is always the same. In the
+next section, you'll see how Doctrine is smart enough to update entries if
+they already exist in MongoDB.
+
+.. tip::
+
+    Doctrine provides a library that allows you to programmatically load testing
+    data into your project (i.e. "fixture data"). For information, see
+    `DoctrineFixturesBundle`_.
+
+Fetching Objects from MongoDB
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Fetching an object back out of MongoDB is even easier. For example, suppose
+you've configured a route to display a specific ``Product`` based on its
+``id`` value:
+
+.. code-block:: php
+
+    public function showAction($id)
+    {
+        $product = $this->get('doctrine_mongodb')
+            ->getRepository('AcmeStoreBundle:Product')
+            ->find($id);
+
+        if (!$product) {
+            throw $this->createNotFoundException('No product found for id '.$id);
+        }
+
+        // do something, like pass the $product object into a template
+    }
+
+When you query for a particular type of object, you always use what's known
+as its "repository". You can think of a repository as a PHP class whose only
+job is to help you fetch objects of a certain class. You can access the
+repository object for a document class via:
+
+.. code-block:: php
+
+    $repository = $this->get('doctrine_mongodb')
+        ->getManager()
+        ->getRepository('AcmeStoreBundle:Product');
+
+.. note::
+
+    The ``AcmeStoreBundle:Product`` string is a shortcut you can use anywhere
+    in Doctrine instead of the full class name of the document (i.e. ``Acme\StoreBundle\Document\Product``).
+    As long as your document lives under the ``Document`` namespace of your bundle,
+    this will work.
+
+Once you have your repository, you have access to all sorts of helpful methods:
+
+.. code-block:: php
+
+    // query by the identifier (usually "id")
+    $product = $repository->find($id);
+
+    // dynamic method names to find based on a column value
+    $product = $repository->findOneById($id);
+    $product = $repository->findOneByName('foo');
+
+    // find *all* products
+    $products = $repository->findAll();
+
+    // find a group of products based on an arbitrary column value
+    $products = $repository->findByPrice(19.99);
+
+.. note::
+
+    Of course, you can also issue complex queries, which you'll learn more
```

Comment:
In Symfony Docs we have removed words like "Of course", "simply", "obviously", etc. because they are hostile to newcomers. Maybe you want to do the same here and remove "of course".

Reply:
I'll create separate issue to go through all docs, don't want to do too many changes as they're hard to spot.